File: HCM_2122/Bai2/VIECNHA.py
# Đọc dữ liệu từ file VIECNHA.INP
with open("VIECNHA.INP", "r") as file_in:
    T = int(file_in.readline().strip())  # Thời gian giới hạn
    C = int(file_in.readline().strip())  # Số lượng việc nhà

    # Danh sách thời gian cho các công việc
    tasks = [int(file_in.readline().strip()) for _ in range(C)]

# Sắp xếp thời gian các công việc theo thứ tự tăng dần
tasks.sort()

# Khởi tạo biến đếm và tổng thời gian
total_time = 0
count = 0

# Thực hiện các công việc cho đến khi hết thời gian
for task in tasks:
    if total_time + task <= T:
        total_time += task
        count += 1
    else:
        break

# Ghi kết quả ra file VIECNHA.OUT
with open("VIECNHA.OUT", "w") as file_out:
    file_out.write(str(count) + "\n")

File: THCS/2021_2022/HCM_2122/Bai2/VIECNHA.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <fstream>
using namespace std;

int main() {
    // Đọc dữ liệu từ file VIECNHA.INP
    ifstream inputFile("VIECNHA.INP");
    ofstream outputFile("VIECNHA.OUT");

    if (!inputFile || !outputFile) {
        cerr << "Không thể mở file." << endl;
        return 1;
    }

    int T, C;
    inputFile >> T >> C;

    // Danh sách thời gian cho các công việc
    vector<int> tasks(C);
    for (int i = 0; i < C; i++) {
        inputFile >> tasks[i];
    }

    // Sắp xếp thời gian các công việc theo thứ tự tăng dần
    sort(tasks.begin(), tasks.end());

    // Khởi tạo biến đếm và tổng thời gian
    int totalTime = 0;
    int count = 0;

    // Thực hiện các công việc cho đến khi hết thời gian
    for (int task : tasks) {
        if (totalTime + task <= T) {
            totalTime += task;
            count++;
        } else {
            break;
        }
    }

    // Ghi kết quả ra file VIECNHA.OUT
    outputFile << count << endl;

    inputFile.close();
    outputFile.close();

    return 0;
}